lib/gumboot/shared_examples/foreign_keys.rb

```diff
@@ -0,0 +1,71 @@
+conn = ActiveRecord::Base.connection
+
+RSpec.shared_examples 'Foreign Keys' do
+  describe 'DB Supports Foreign Keys', if: conn.supports_foreign_keys? do
+    context 'Permissions Foreign Keys' do
+      it 'should have a foreign key' do
+        expect(table_has_fk('permissions', 'role_id')).to be true
+      end
+      it 'should have a foreign key that points to the correct table' do
+        expect(fk_exists_between('permissions', 'roles')).to be true
+      end
```

Comment:
L6 - 11 is essentially duplicated in every context here.

Please pull this out into a shared example so the code appears exactly one, and re-use it.